src/Symfony/Bundle/WebProfilerBundle/Controller/RouterController.php

```diff
@@ -99,4 +103,26 @@ private function getTraces(RequestDataCollector $request, $method)
 
         return $matcher->getTracesForRequest($traceRequest);
     }
+
+    /**
+     * Replaces scalars wrapped as Data instances by their actual values and removes other values.
```

Comment:
but you also get rid of cases where the Data instance contains a non-scalar value. this is wrong in several cases:

- when it wraps ``null`` (which is *not* a scalar)
- when it wraps an array (and yes, having nested arrays in request attributes is supported today. It would be even worse for GET and POST but they are not filtered here)